### ISSUE TITLE
Install footprint has increased considerably in lxc-4.0.4

### ISSUE THREAD

**Issue by sparlane** [lxc/lxc#3523]
When we upgrade from lxc-4.0.1 to lxc-4.0.4 our initrd grew by 6.8MB (this represents >10% of the original size)

We build from source with the following configuration:
       '--sysconfdir=/etc',
        '--disable-doc',
        '--disable-api-docs',
        '--disable-apparmor',
        '--disable-selinux',
        '--enable-seccomp',
        '--disable-examples',
        '--disable-bash',
        '--with-init-script=',
        '--with-global-conf=/etc/lxc/lxc.conf',
        '--with-distro=unknown',
GCC is 4.7.4 and 4.9.3, the initrd has the binaries stripped and then gziped

before strip is run, the lxc-* binaries are all 3.4M in 4.0.4 (in 4.0.3 they are 109->831K, most were ~120K)
after strip is run, most are now 951K (in 4.0.3 mostly 68K)

@brauner, I think this is a side effect of #3492

**Comment by stgraber** [comment]
Yeah, I'd expect this to be a side effect of removing a whole bunch of internal symbols from our shared library.
I still think it's the right move overall, we shouldn't have our public library include a whole bunch of internal functions with tools using those symbols and so breaking should the library alone be upgraded/downgrade.

One option to get the best of both worlds would be to have an internal shared library which the tools use and which isn't versioned and must always be from the same build as the tools. This would let the individual binaries go back to being tiny while also not having our public library be used for this.

**Comment by sparlane** [comment]
How can I help to resolve this so we can have both?

Also what is the timing of lxc-4.0.5 likely to be?

Thanks

**Comment by stgraber** [comment]
No clear timing at this point. We release when we feel we have enough important fixes accumulated.

As for adding an internal shared library, that would be a question for @brauner

**Comment by brauner** [comment]
> How can I help to resolve this so we can have both?
> 
> Also what is the timing of lxc-4.0.5 likely to be?
> 
> Thanks

Hm, I'm happy to merge something that would fix this. Do you feel up for this task @sparlane?

**Comment by mihalicyn** [comment]
This is something I tried to fix in https://github.com/lxc/lxc/pull/4401

But there is still some work to do.

**Comment by mihalicyn** [comment]
Let's close this as we have similar issue filled here https://github.com/lxc/lxc/issues/4251
